fileuploader-api/main.py

```
@@ -0,0 +1,489 @@
+                urn_checklist = add_error(urn_checklist, item_urn, "aspect is not a dataset or container aspect")
+                continue
+            if item_urn not in urn_checklist:
+                urn_checklist = add_new_urn(urn_checklist, item_urn, user_id, entityType)                                                
+            if aspectName=="ownership":
+                ownership_data_aspect = check_urn.value
+                ownership_data = ownership_data_aspect.owners
+                owners_in_data = {item.owner: item.type for item in ownership_data}
+                # owners_in_data = [item.get("owner","") for item in ownership_data_aspect.get("ownership",{})]
+                if f"urn:li:corpuser:{user_id}" not in owners_in_data:
+                    print("not in owners list")
+                    urn_checklist = update_retain_ownership(urn_checklist, item_urn, False, owners_in_data)
+                else:
+                    urn_checklist = update_retain_ownership(urn_checklist, item_urn, True, owners_in_data)
+        else:
+            urn_checklist = add_error(urn_checklist, "unable_to_infer", f"Item {i} in file is invalid object")
+    pre_ingest_check = True # assume all is ok until otherwise
+    for check_urn in urn_checklist:
+        # for new assets only; to add uploader as owner
+            # if "retain_ownership" has no value then need to create ownership aspect 
+        if not urn_checklist[check_urn].get("retains_ownership", False):
+            urn_checklist[check_urn]["add_ownership"] = True
+        if urn_checklist[check_urn].get("existing_entity",True): 
+            # if existing entity and existing ownership state is false, reject
+            if not urn_checklist[check_urn].get("ownership", True):
+                pre_ingest_check = False
+                urn_checklist = add_error(urn_checklist, check_urn, "does not own dataset")                
+                continue            
```

should just fail fast instead of wasting processing power of the server to look through the rest of the json. When an error occurs, whether it is an exception or a validation error or not, the service should stop processing and return that error status to the consumer. This will reserve processing power for the server.